#This compose file adds Pydio https://pydio.com/en/pydio-8-overview to your stack
#according to the maintainer, you will have to manually input the database hostname
#database name, and database username/password.  Despite what it says, use the
#hostname pydio-db (or whatever you name the pydio-db service below).
#https://store.docker.com/community/images/linuxserver/pydio
version: '3.5'
services:
  reverse-proxy:
    networks:
      - pydio
  mail-relay:
    networks:
      - pydio
  pydio:
    image: linuxserver/pydio:latest
    container_name: pydio
    environment:
      - TZ=${TZ}
    volumes:
      - "pydio-config:/config"
      - "pydio-data:/data"
    labels:
      - traefik.enable=true
      - traefik.port=80
      - traefik.frontend.rule=Host:${PYDIO_DOMAIN:?Please copy template.env to .env and provide provide a value for PYDIO_DOMAIN}
      - traefik.docker.network=pydio
      - traefik.default.protocol=http
      - com.ouroboros.enable=true
    expose:
      - "80"
    networks:
      - pydio
    restart: unless-stopped
    depends_on:
      - pydio-db
  pydio-db:
    image: mariadb:latest
    container_name: pydio-db
    environment:
      - MYSQL_DATABASE=pydio
      - MYSQL_USER=${PYDIO_DB_USERNAME:-pydio}
      - MYSQL_PASSWORD=${PYDIO_DB_PASSWORD:?Please copy template.env to .env and provide provide a value for PYDIO_DB_PASSWORD}
      - MYSQL_ROOT_PASSWORD=${PYDIO_DB_ROOT_PASSWORD:?Please copy template.env to .env and provide provide a value for PYDIO_DB_ROOT_PASSWORD}
    volumes:
      - "pydio-database-data:/var/lib/mysql"
    labels:
      - "traefik.enable=false"
      - "com.ouroboros.enable=true"
    restart: unless-stopped
    networks:
      - pydio

networks:
  pydio:

volumes:
  pydio-config:
  pydio-data:
  pydio-database-data:


